app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,47 @@
+class Api::V1::ReservationsController < ApplicationController
+  def index
+    @reservations = User.find_by(username: params[:username]).reservations.includes(:item)
+    render json: {
+      reservations:
+        @reservations.map do |reservation|
```

Comment:
I would avoid using array methods with data. If there are hundreds (or even thousands) of reservations for a user then this will make your application very slow. Now multiply that to hundreds if not thousands of users concurrently making a request to your Reservations index end point.

A more proper way to handle this is to use ActiveRecord queries (such as where). And in fact we shouldn't be doing it here in the controller but on the model.